examples/publisher_simple.rs

```diff
@@ -14,7 +14,7 @@ struct CounterTopic {
     counter: u32,
 }
 
-unsafe impl POD for CounterTopic {}
+impl POD for CounterTopic {}
```

Comment:
Not sure about this either. I know that it is not nice to have `unsafe` in the code but this marker trait is similar to `Send` and `Sync`. One has to be careful to implement it since the data which can be send via shared memory has some constraints. I think I could use the `Copy` trait as bound but that would restrict the data types more than necessary. Ideally there would be a `CouldBeCopy` trait bound. I would be happy about a solution without `unsafe` but currently one has to be careful when the trait is implemented. Ideas how to solve this in a different way are welcome and highly appreciated.

Reply:
I think right now, this is mostly a documentation issue. `clippy` was complaining about [missing safety doc](clippy::missing_safety_doc). I did not see a need for `POD`  to be `unsafe`, so I removed the qualifier :smile: 

The most elegant solution is probably having some sort of a `iox::pod` attribute macro. This macro would check the constraints and  generate the code for shared memory decoding / encoding and the Trait implementations needed.

Something like

```Rust
#[iox::pod]
pub struct CounterTopic {
...
}
```

So in other words, the idea behind `iox::pod` would be to hide as much of the `iceoryx` protocol / machinery as possible.

Reply:
@adoerr yes, that would be ideal. If you have the knowledge, time and inclination it would be a great contribution. My knowledge regarding proc macros is quite limited but it is on my todo list :smile: